src/utils/MSstatsConverter.cpp

```diff
@@ -0,0 +1,419 @@
+      {
+        LOG_ERROR << "ERROR: ProteinIdentification does not have meta value for original file. Cannot continue!" << std::endl;
+        return ILLEGAL_PARAMETERS;
+      }
+
+      for (auto const & meta_value : protein_identification.getMetaValue(TOPPMSstatsConverter::meta_value_exp_design_key).toStringList())
+      {
+        // Split the identifier by path separator character
+        std::vector< std::string > strs;
+        boost::split(strs, meta_value, boost::is_any_of("/\\"));
+        assert(strs.empty() == false);
+        protid_to_filenames[identifier].insert(strs.back());
+      }
+    }
+
+    // These are placeholder fragment annotations and peptide evidences in case the original ones are empty
+
+    // Placeholder fragment annotation
+    PeptideHit::PeakAnnotation new_peak_ann;
+    new_peak_ann.annotation = TOPPMSstatsConverter::na_string;
+    new_peak_ann.charge = -1;
+    std::vector< PeptideHit::PeakAnnotation > placeholder_fragment_annotations = {new_peak_ann};
+
+    // Placeholder peptide evidence
+    PeptideEvidence new_pep_ev;
+    new_pep_ev.setProteinAccession(TOPPMSstatsConverter::na_string);
+    std::vector< PeptideEvidence > placeholder_peptide_evidences = {new_pep_ev};
+
+    // From the MSstats user guide: endogenous peptides (use "L") or labeled reference peptides (use "H").
+    String isotope_label_type = this->getFlag_(TOPPMSstatsConverter::param_labeled_reference_peptides) ? "H" : "L";
+
+    // Keeps track of precursor charges and accessions of a peptide sequence to avoid duplicate lines in output
+    String previous_sequence = "";
+    std::set< Int > previous_precursor_charges;
+    std::set< String > previous_prot_accs;
+
+    for (auto const & consensus_feature : consensus_map)
+    {
+      Peak2D::IntensityType intensity = consensus_feature.getIntensity();
+      assert(intensity > 0);
+
+      for (auto const & pep_id : consensus_feature.getPeptideIdentifications())
+      {
+        // Get runs that belong to this identifier
+        std::set< Size > runs;
+        assert(protid_to_filenames.find(pep_id.getIdentifier()) != protid_to_filenames.end());
+
+        for (auto const & filename : protid_to_filenames[pep_id.getIdentifier()])
+        {
+          // Test whether the experimental design specifies the encountered filename
+          if (edkey_to_rowindex.find(filename) == edkey_to_rowindex.end())
+          {
+            LOG_ERROR << "ERROR: Experimental design does not contain information on file " << filename << ". Cannot continue!" << std::endl;
+            return ILLEGAL_PARAMETERS;
+          }
+          std::set< Size > const & indices = edkey_to_rowindex[filename];
+          runs.insert(indices.begin(), indices.end());
+        }
+
+        for (auto const & pep_hit : pep_id.getHits())
+        {
+          std::vector< PeptideHit::PeakAnnotation > const & original_fragment_annotations = pep_hit.getPeakAnnotations();
```

Comment:
Also, the examples in https://github.com/OpenMS/OpenMS/wiki/Coding-conventions follow this rule.